tests/snap_test.go

```diff
@@ -0,0 +1,244 @@
+	if !useGPIOMock() || !utils.LocalServiceSnap() {
+		return snapPath, nil
+	}
+
+	// check if gpio mock is enabled AND the service is running locally
+	// Run gpio mockup script
+	_, stderr, err := utils.Exec(nil, "./gpio-mock.sh")
+	if err != nil {
+		return snapPath, fmt.Errorf("Failed to run gpio mockup script %s: %s", stderr, err)
+	}
+
+	// authorize gpio mock
+	newPath, err := authorizeGpioMock(snapPath)
+	if err != nil {
+		return snapPath, err
+	}
+
+	return newPath, nil
+}
+
+func useGPIOMock() bool {
+	return os.Getenv(gpioChipMock) == "true"
+}
+
+func getMockGPIO() (string, error) {
+	gpioChipNumber, stderr, err := utils.Exec(nil, "ls /dev/gpiochip* | sort -n | tail -n 1 | cut -d'p' -f3")
+	if err != nil || stderr != "" {
+		return "", fmt.Errorf("failed to get mock gpio chip number, Error %s: %s", stderr, err)
+	}
+	gpioChipNumber = strings.TrimSpace(gpioChipNumber)
+	gpioChipNumber = strings.Trim(gpioChipNumber, "\n")
+	return gpioChipNumber, nil
+}
+
+func authorizeGpioMock(path string) (string, error) {
+	utils.Exec(nil, "rm -rf squashfs-root")
+
+	_, stderr, err := utils.Exec(nil, "unsquashfs "+path)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	_, stderr, err = utils.Exec(nil, sedAuthorizeMockRead)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	_, stderr, err = utils.Exec(nil, sedAuthorizeMockGPIOsChips)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	directory := filepath.Dir(path)
+	newPath := directory + "/mod_matter-pi-gpio-commander.snap"
+	_, stderr, err = utils.Exec(nil, "mksquashfs squashfs-root "+newPath)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	utils.Exec(nil, "rm -rf squashfs-root")
+	return newPath, nil
+}
+
+func setupGPIO() error {
+	var err error
+	// The GPIO_MOCKUP takes precedence over the specific GPIO_CHIP and GPIO_LINE
+	if useGPIOMock() && utils.LocalServiceSnap() {
+		utils.SnapSet(nil, snapMatterPiGPIO, "gpiochip-validation", "false")
+
+		gpioChip, err = getMockGPIO()
+		if err != nil {
+			return fmt.Errorf("failed to get mock gpio chip number: %s", err)
+		}
+		gpioLine = "4"
+
+		log.Printf("[TEST] Using mockup gpio: /dev/gpiochip%s", gpioChip)
+		log.Printf("[TEST] Using default gpio-mock line: %s", gpioLine)
+	}
+
+	utils.SnapSet(nil, snapMatterPiGPIO, "gpiochip", gpioChip)
+	utils.SnapSet(nil, snapMatterPiGPIO, "gpio", gpioLine)
+
+	return nil
+}
+
+func TestBlinkOperation(t *testing.T) {
+	// test blink operation
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+
+	// Kill the process after 15 seconds
+	go func() {
+		time.Sleep(10 * time.Second)
+		utils.Exec(t, `sudo pkill -f "/snap/matter-pi-gpio-commander/x1/bin/test-blink"`)
+	}()
```

Comment:
Add link to issue or explain why:
```suggestion
	// Force kill the process
	// This is required because on GH Runner, the cancellation only kills the parent process
	go func() {
		time.Sleep(10 * time.Second)
		utils.Exec(t, `sudo pkill -f "/snap/matter-pi-gpio-commander/x1/bin/test-blink"`)
	}()
```